src/nvim/diff.c

```diff
@@ -2627,26 +2620,67 @@ void ex_diffgetput(exarg_T *eap)
     }
   }
 
+  diffgetput(eap->addr_count, idx_cur, idx_from, idx_to, eap->line1, eap->line2);
+
+  // restore curwin/curbuf and a few other things
+  if (eap->cmdidx != CMD_diffget) {
+    // Syncing undo only works for the current buffer, but we change
+    // another buffer.  Sync undo if the command was typed.  This isn't
+    // 100% right when ":diffput" is used in a function or mapping.
+    if (KeyTyped) {
+      u_sync(false);
+    }
+    aucmd_restbuf(&aco);
+  }
+
+theend:
+  diff_busy = false;
+
+  if (diff_need_update) {
+    ex_diffupdate(NULL);
+  }
+
+  // Check that the cursor is on a valid character and update its
+  // position.  When there were filler lines the topline has become
+  // invalid.
+  check_cursor();
+  changed_line_abv_curs();
+
+  if (diff_need_update) {
+    // redraw already done by ex_diffupdate()
+    diff_need_update = false;
+  } else {
+    // Also need to redraw the other buffers.
+    diff_redraw(false);
+    apply_autocmds(EVENT_DIFFUPDATED, NULL, NULL, false, curbuf);
+  }
+}
+
+// Apply diffget/diffput to buffers and difflbocks
```

Comment:
```suggestion
```suggestion
/// Apply diffget/diffput to buffers and diffblocks
///
/// @param idx_cur   index of "curbuf" in the list of diff buffers before aucmd_prepbuf()
/// @param idx_from  index of the buffer to read from in the list of diff buffers
/// @param idx_to    index of the buffer to modify in the list of diff buffers
```
```

Reply:
There is a mistake in the suggestion. Please revert apply it again.